File: app/src/main/java/sk/spacecode/silentkosice/RecordActivity.java
package sk.spacecode.silentkosice;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Message;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import java.io.File;


public class RecordActivity extends AppCompatActivity {

    float volume = 10000;
    private MyMediaRecorder mRecorder;
    private TextView actualDB;
    private ImageView profileText;
    private ImageView progressCircle2;
    private ImageView imageButton;
    private ImageView imageAudio;
    private static final int msgWhat = 0x1001;
    private static final int refreshTime = 100;

    private boolean recording = false;
    @SuppressLint("HandlerLeak")
    private Handler handler = new Handler() {
        @Override
        public void handleMessage(Message msg) {
            super.handleMessage(msg);
            if (this.hasMessages(msgWhat)) {
                return;
            }
            volume = mRecorder.getMaxAmplitude();
            if (volume > 0 && volume < 1000000) {
                World.setDbCount(20 * (float) (Math.log10(volume)));
                actualDB.setText((int) World.dbCount + " dB");

                imageButton.getLayoutParams().height = (((int) World.dbCount * 100) / 140) * 12;
                imageButton.getLayoutParams().width = (((int) World.dbCount * 100) / 140) * 12;
                imageAudio.getLayoutParams().height = (((int) World.dbCount * 100) / 140) * 12;
                imageAudio.getLayoutParams().width = (((int) World.dbCount * 100) / 140) * 12;
                progressCircle2.getLayoutParams().height = (((int) World.dbCount * 100) / 90) * 11;
                progressCircle2.getLayoutParams().width = (((int) World.dbCount * 100) / 90) * 11;
            }

            handler.sendEmptyMessageDelayed(msgWhat, refreshTime);
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_record);
        mRecorder = new MyMediaRecorder();

        progressCircle2 = findViewById(R.id.imageView_circleProgress2);
        imageButton = findViewById(R.id.imageView_circleProgress);
        imageAudio = findViewById(R.id.imageView_audio_image);

        actualDB = findViewById(R.id.textView_currentDB);
        profileText = findViewById(R.id.profile_button);

        File file = FileUtil.createFile("temp.amr");
        if (file != null) {
            Log.v("file", "file =" + file.getAbsolutePath());
            startRecord(file);
        } else {
            Toast.makeText(getApplicationContext(), "Something went terribly wrong", Toast.LENGTH_LONG).show();
        }

        profileText.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {
                Intent myIntent = new Intent(RecordActivity.this, ProfileActivity.class);
                RecordActivity.this.startActivity(myIntent);
            }
        });

        imageButton.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {

                if(!recording){

                    recording = true;
                    File file = FileUtil.createFile("temp.amr");
                    if (file != null) {
                        Log.v("file", "file =" + file.getAbsolutePath());
                        startRecord(file);
                    } else {
                        Toast.makeText(getApplicationContext(), "Something went terribly wrong", Toast.LENGTH_LONG).show();
                    }
                }else {
                    progressCircle2.getLayoutParams().height = 340;
                    progressCircle2.getLayoutParams().width = 340;
                    mRecorder.delete();
                    handler.removeMessages(msgWhat);
                    actualDB.setText("Not recording");

                }
            }
        });

        GPSHelper gps = new GPSHelper(this);
        gps.getMyLocation();
        Log.e("TVOJAMATKA", gps.getLatitude());
        Log.e("TVOJAMATKA", gps.getLongitude());

    }
    
    private void startListenAudio() {
        handler.sendEmptyMessageDelayed(msgWhat, refreshTime);
    }


    public void startRecord(File fFile){
        try{
            mRecorder.setMyRecAudioFile(fFile);
            if (mRecorder.startRecorder()) {
                startListenAudio();
            }else{
                Toast.makeText(this, "Recording already started", Toast.LENGTH_SHORT).show();
            }
        }catch(Exception e){
            Toast.makeText(this, "Something went terribly wrong", Toast.LENGTH_SHORT).show();
            e.printStackTrace();
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
    }


    @Override
    protected void onPause() {
        super.onPause();
        mRecorder.delete();
        handler.removeMessages(msgWhat);
    }

    @Override
    protected void onDestroy() {
        handler.removeMessages(msgWhat);
        mRecorder.delete();
        super.onDestroy();
    }
}

File: app/src/main/java/sk/spacecode/silentkosice/RecordActivity.kt
package sk.spacecode.silentkosice

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.support.v7.app.AppCompatActivity
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import kotlinx.android.synthetic.main.activity_record.*


class RecordActivity : AppCompatActivity() {

    private var i = 0
    var handler: Handler? = null

    private lateinit var database: DatabaseReference

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_record)
//        output = externalCacheDir?.absolutePath + "/${convertDateToFilename()}.mp3"
//        mediaRecorder.setAudioSource(MediaRecorder.AudioSource.MIC)
//        mediaRecorder.setOutputFormat(MediaRecorder.OutputFormat.MPEG_4)
//        mediaRecorder.setAudioEncoder(MediaRecorder.AudioEncoder.AAC)
//        mediaRecorder.setOutputFile(output)
//
        button_start_recording.setOnClickListener {
//            try {
//                mediaRecorder.prepare()
//                mediaRecorder.start()
//                state = true
//                Toast.makeText(this, "Recording started!", Toast.LENGTH_SHORT).show()
//            } catch (e: IllegalStateException) {
//                e.printStackTrace()
//            } catch (e: IOException) {
//                e.printStackTrace()
//            }
            val tsLong = System.currentTimeMillis() / 1000
            pushDataRecord(tsLong.toString(), "70","+4210902130280", "37.90909", "37.90909")

        }
//
//        button_stop_recording.setOnClickListener {
//            if (state) {
//                mediaRecorder.stop()
//                mediaRecorder.release()
//                state = false
//            } else {
//                Toast.makeText(this, "You are not recording right now!", Toast.LENGTH_SHORT).show()
//            }
//        }


        database = FirebaseDatabase.getInstance().reference


//
//        handler = Handler()
//        var mRecorder = MediaRecorder()
//        mRecorder.setAudioSource(MediaRecorder.AudioSource.MIC)
//        mRecorder.setOutputFormat(MediaRecorder.OutputFormat.THREE_GPP)
//        mRecorder.setAudioEncoder(MediaRecorder.AudioEncoder.AMR_NB)
//        mRecorder.setOutputFile("/dev/null")
//        try {
//            mRecorder.prepare()
//        } catch (e: IllegalStateException) {
//            e.printStackTrace()
//        } catch (e: IOException) {
//            e.printStackTrace()
//        }
//
//        mRecorder.start()
//
//        handler!!.postDelayed(runnable, 0)

        textView_profile.setOnClickListener {
            val profileIntent = Intent(this@RecordActivity, ProfileActivity::class.java)
            this@RecordActivity.startActivity(profileIntent)
        }
    }

    private fun pushDataRecord(timestamp: String, decibel: String, userNumber: String?, lat: String?, long: String?) {
        val data = DecibelData(timestamp, decibel, userNumber, lat, long)
        val newRef = database.child(userNumber!!).child("records").child(timestamp).setValue(data)
    }


    var runnable: Runnable = object : Runnable {
        override fun run() {
            i++
            textView_currentDB.text = (i.toString())
            handler!!.postDelayed(this, 0)
        }
    }


}
